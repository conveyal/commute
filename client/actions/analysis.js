import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
