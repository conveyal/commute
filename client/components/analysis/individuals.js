import humanizeDuration from 'humanize-duration'
import {Browser} from 'leaflet'
import React, {Component, PropTypes} from 'react'
import {Button, Col, Grid, Row} from 'react-bootstrap'
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table'
import {Map as LeafletMap, TileLayer} from 'react-leaflet'
import {Link} from 'react-router'

import Icon from '../icon'
import {humanizeDistance} from '../../utils/components'
import {settings} from '../../utils/env'

export default class Individuals extends Component {
  static propTypes = {
    // dispatch
    deleteAnalysis: PropTypes.func,

    // props
    analysis: PropTypes.object.isRequired,
    commutersById: PropTypes.object.isRequired,
    groupName: PropTypes.string.isRequired,
    organizationId: PropTypes.string.isRequired,
    siteName: PropTypes.string.isRequired
  }

  constructor (props) {
    super(props)
    this.state = {
      position: settings.map.focus,
      zoom: 8
    }
  }

  _costRenderer = (cell, row) => {
    return `$${row.mostLikely.cost}`
  }

  _distanceRenderer = (cell, row) => {
    return humanizeDistance(row.mostLikely.distance)
  }

  _durationRenderer = (cell, row) => {
    return humanizeDuration(row.mostLikely.time * 60 * 1000)
  }

  _modeRenderer = (cell, row) => {
    return row.mostLikely.mode
  }

  _nameRenderer = (cell, row) => {
    return this.props.commutersById[row.commuterId].name
  }

  _onRowSelect = (row, isSelected) => {
    // TODO: show on map
    console.log(row)
    console.log('selected: ' + isSelected)
  }

  _ridematchesRenderer = (cell, row) => {
    return row.mostLikely.ridematches
  }

  render () {
    const {id, name} = this.props.analysis
    const {analysis, groupName, organizationId, siteName} = this.props
    const {position, zoom} = this.state
    return (
      <Grid>
        <Row className='individuals-header'>
          <Col xs={12}>
            <h3>
              <span>{name}</span>
              <Button className='pull-right'>
                <Link to={`/organizations/${organizationId}/analysis/${id}`}>
                  <Icon type='arrow-left' />
                  <span>Back</span>
                </Link>
              </Button>
            </h3>
            <h3>Individual Commutes</h3>
          </Col>
          <Col xs={12} md={6}>
            <h4>
              <strong>Site:</strong>
              <span>{siteName}</span>
            </h4>
          </Col>
          <Col xs={12} md={6}>
            <h4>
              <strong>Group:</strong>
              <span>{groupName}</span>
            </h4>
          </Col>
        </Row>
        <Row>
          <Col xs={8} style={{height: '400px'}}>
            <LeafletMap center={position} zoom={zoom}>
              <TileLayer
                url={Browser.retina &&
                  process.env.LEAFLET_RETINA_URL
                  ? process.env.LEAFLET_RETINA_URL
                  : process.env.LEAFLET_TILE_URL}
                attribution={process.env.LEAFLET_ATTRIBUTION}
                />
            </LeafletMap>
          </Col>
        </Row>
        <Row className='individuals-content'>
          <Col xs={12}>
            <h3>Individual Trips</h3>
            <BootstrapTable
              data={analysis.trips}
              selectRow={{
                mode: 'radio',
                clickToSelect: true,
                bgColor: 'rgb(238, 193, 213)',
                onSelect: this._onRowSelect
              }}
              >
              <TableHeaderColumn dataField='commuterId' isKey hidden />
              <TableHeaderColumn dataFormat={this._nameRenderer}>Name</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._modeRenderer}>Mode</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._durationRenderer}>Duration</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._distanceRenderer}>Distance</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._costRenderer}>Cost</TableHeaderColumn>
              <TableHeaderColumn dataFormat={this._ridematchesRenderer}>Ridematches</TableHeaderColumn>
            </BootstrapTable>
          </Col>
        </Row>
      </Grid>
    )
  }
}
