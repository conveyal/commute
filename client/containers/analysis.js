import {connect} from 'react-redux'

import analysisActions from '../actions/analysis'
import commuterActions from '../actions/commuter'
import _histogram from '../components/analysis/histogram'
import _individuals from '../components/analysis/individuals'
import _possibilities from '../components/analysis/possibilities'
import _summary from '../components/analysis/summary'

function mapStateToProps (state, props) {
  const {analysis, commuter, group, site} = state
  const currentAnalysis = analysis[props.params.analysisId]
  return {
    analysis: currentAnalysis,
    commuterStore: commuter,
    groupName: group[currentAnalysis.groupId].name,
    site: site[currentAnalysis.siteId]
  }
}

function mapDispatchToProps (dispatch) {
  return {
    deleteAnalysis: (opts) => dispatch(analysisActions.delete(opts)),
    loadAnalysis: (opts) => dispatch(analysisActions.loadOne(opts)),
    loadCommuters: (opts) => dispatch(commuterActions.loadMany(opts))
  }
}

export const Histogram = connect(mapStateToProps)(_histogram)
export const Individuals = connect(mapStateToProps, mapDispatchToProps)(_individuals)
export const Possibilities = connect(mapStateToProps)(_possibilities)
export const Summary = connect(mapStateToProps, mapDispatchToProps)(_summary)
