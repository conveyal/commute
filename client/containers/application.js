import {connect} from 'react-redux'
import {refreshUser} from '@conveyal/woonerf/auth0'

import Application from '../components/application'

function mapStateToProps (state) {
  return {
    userIsLoggedIn: !!state.user
  }
}

function mapDispatchToProps (dispatch) {
  return {
    refreshUserToken: () => refreshUser(dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Application)
