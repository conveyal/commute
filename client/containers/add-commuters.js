import {connect} from 'react-redux'

import {appendToGroup, createGroup} from '../actions/group'
import AddCommuters from '../components/add-commuters'

function mapStateToProps (state, props) {
  const {organization} = state
  const {params} = props
  const currentOrganizationId = params.organizationId
  const output = {
    organizationId: currentOrganizationId
  }
  if (params && params.groupId) {
    const currentOrganization = organization.organizationsById[currentOrganizationId]
    const currentGroup = currentOrganization.groupsById[params.groupId]
    Object.assign(output, {
      appendMode: true,
      organizationId: currentOrganizationId,
      group: currentGroup
    })
  }
  return output
}

function mapDispatchToProps (dispatch, props) {
  return {
    create: (group, organizationId) =>
      dispatch(createGroup(group, organizationId)),
    append: (commuters, groupId, organizationId) =>
      dispatch(appendToGroup(commuters, groupId, organizationId))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddCommuters)
