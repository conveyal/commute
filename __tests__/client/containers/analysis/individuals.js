/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {makeMockStore, mockStores} from '../../../test-utils/mock-data.js'
import '../../../test-utils/mock-leaflet'

import {Individuals} from '../../../../client/containers/analysis'

const mockStore = makeMockStore(mockStores.complexOrganization)

describe('Container > Analysis > Individuals', () => {
  it('Individuals View loads (base case)', () => {
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Individuals
          params={{organizationId: '2', analysisId: '1'}}
          />
      </Provider>
      , {
        attachTo: document.getElementById('test')
      }
    )
    expect(mountToJson(tree.find('.individuals-header'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.individuals-content'))).toMatchSnapshot()
  })

  /* it('Select commuter', () => {
    // Given a logged-in user is viewing the Individual Commuter Analysis View
    // When the user selects a row of a commuter in the table
    // Then the CarFreeAtoZ map and sidebar should get updated with the commute possibilities for the selected commuter
  }) */
})
