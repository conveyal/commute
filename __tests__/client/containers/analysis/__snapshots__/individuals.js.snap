exports[`Container > Analysis > Individuals Individuals View loads (base case) 1`] = `
<div
  className="individuals-header row">
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}>
    <div
      className="col-xs-12">
      <h3>
        <span>
          An Analysis
        </span>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="pull-right"
          disabled={false}>
          <button
            className="pull-right btn btn-default"
            disabled={false}
            type="button">
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/organizations/2/analysis/1">
              <a
                onClick={[Function]}
                style={Object {}}>
                <Icon
                  type="arrow-left">
                  <i
                    className="fa fa-arrow-left fa-fw " />
                </Icon>
                <span>
                  Back
                </span>
              </a>
            </Link>
          </button>
        </Button>
      </h3>
      <h3>
        Individual Commutes
      </h3>
    </div>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={6}
    xs={12}>
    <div
      className="col-md-6 col-xs-12">
      <h4>
        <strong>
          Site:
        </strong>
        <span>
          Acme Corp
        </span>
      </h4>
    </div>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={6}
    xs={12}>
    <div
      className="col-md-6 col-xs-12">
      <h4>
        <strong>
          Group:
        </strong>
        <span>
          Mock Group
        </span>
      </h4>
    </div>
  </Col>
</div>
`;

exports[`Container > Analysis > Individuals Individuals View loads (base case) 2`] = `
<div
  className="individuals-content row">
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}>
    <div
      className="col-xs-12">
      <h3>
        Individual Trips
      </h3>
      <BootstrapTable
        bodyContainerClass={null}
        bordered={true}
        cellEdit={
          Object {
            "__onCompleteEdit__": [Function],
            "afterSaveCell": undefined,
            "beforeSaveCell": undefined,
            "blurToSave": false,
            "mode": "none",
          }
        }
        columnFilter={false}
        condensed={false}
        containerClass={null}
        csvFileName="spreadsheet.csv"
        data={
          Array [
            Object {
              "bike": Object {
                "cost": 0.12,
                "distance": 23,
                "polygon": "encoded",
                "time": 2345,
              },
              "car": Object {
                "cost": 6.78,
                "distance": 18,
                "polygon": "encoded",
                "time": 1234,
              },
              "commuterId": "1",
              "mostLikely": Object {
                "cost": 3.45,
                "distance": 30,
                "mode": "transit",
                "polygon": "encoded",
                "time": 3456,
              },
              "transit": Object {
                "cost": 3.45,
                "distance": 30,
                "polygon": "encoded",
                "time": 3456,
              },
              "walk": Object {
                "cost": 0,
                "distance": 19,
                "polygon": "encoded",
                "time": 6789,
              },
            },
          ]
        }
        deleteRow={false}
        exportCSV={false}
        fetchInfo={
          Object {
            "dataTotalSize": 0,
          }
        }
        headerContainerClass={null}
        height="100%"
        hover={false}
        ignoreSinglePage={false}
        insertRow={false}
        multiColumnSearch={false}
        options={
          Object {
            "afterColumnFilter": undefined,
            "afterDeleteRow": undefined,
            "afterInsertRow": undefined,
            "afterSearch": undefined,
            "afterTableComplete": undefined,
            "clearSearch": false,
            "closeText": "Close",
            "defaultSearch": "",
            "defaultSortName": undefined,
            "defaultSortOrder": undefined,
            "deleteText": "Delete",
            "exportCSVText": "Export to CSV",
            "firstPage": "<<",
            "handleConfirmDeleteRow": undefined,
            "hideSizePerPage": false,
            "ignoreEditable": false,
            "insertText": "New",
            "lastPage": ">>",
            "nextPage": ">",
            "noDataText": undefined,
            "onMouseEnter": undefined,
            "onMouseLeave": undefined,
            "onRowClick": undefined,
            "onRowDoubleClick": undefined,
            "onRowMouseOut": undefined,
            "onRowMouseOver": undefined,
            "onSizePerPageList": undefined,
            "page": undefined,
            "pageStartIndex": undefined,
            "paginationShowsTotal": false,
            "paginationSize": 5,
            "prePage": "<",
            "saveText": "Save",
            "searchDelayTime": undefined,
            "sizePerPage": undefined,
            "sizePerPageList": Array [
              10,
              25,
              30,
              50,
            ],
            "sortIndicator": true,
            "sortName": undefined,
            "sortOrder": undefined,
          }
        }
        pagination={false}
        printable={false}
        search={false}
        selectRow={
          Object {
            "bgColor": "rgb(238, 193, 213)",
            "clickToSelect": true,
            "mode": "radio",
            "onSelect": [Function],
          }
        }
        striped={false}
        tableBodyClass={null}
        tableContainerClass={null}
        tableHeaderClass={null}
        trClassName="">
        <div
          className="react-bs-table-container">
          <div
            className="react-bs-table"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "height": "100%",
                "maxHeight": undefined,
              }
            }>
            <TableHeader
              bordered={true}
              condensed={false}
              headerContainerClass={null}
              isFiltered={false}
              isSelectAll={false}
              onSelectAllRow={[Function]}
              onSort={[Function]}
              rowSelectType="radio"
              sortIndicator={true}
              tableHeaderClass={null}>
              <div
                className="react-bs-container-header table-header-wrapper">
                <table
                  className="table table-hover table-bordered">
                  <thead>
                    <tr>
                      <SelectRowHeaderColumn>
                        <th
                          style={
                            Object {
                              "textAlign": "center",
                            }
                          } />
                      </SelectRowHeaderColumn>
                      <TableHeaderColumn
                        className=""
                        columnClassName=""
                        columnTitle={false}
                        dataAlign="left"
                        dataField="commuterId"
                        dataSort={false}
                        editable={true}
                        filterFormatted={false}
                        headerTitle={true}
                        hidden={true}
                        hiddenOnInsert={false}
                        isKey={true}
                        onSort={[Function]}
                        searchable={true}
                        sortIndicator={true}
                        width={null}>
                        <th
                          className=""
                          onClick={[Function]}
                          style={
                            Object {
                              "display": "none",
                              "textAlign": "left",
                            }
                          }>
                          <div
                            onClick={[Function]} />
                        </th>
                      </TableHeaderColumn>
                      <TableHeaderColumn
                        className=""
                        columnClassName=""
                        columnTitle={false}
                        dataAlign="left"
                        dataFormat={[Function]}
                        dataSort={false}
                        editable={true}
                        filterFormatted={false}
                        headerTitle={true}
                        hidden={false}
                        hiddenOnInsert={false}
                        isKey={false}
                        onSort={[Function]}
                        searchable={true}
                        sortIndicator={true}
                        width={null}>
                        <th
                          className=""
                          onClick={[Function]}
                          style={
                            Object {
                              "display": null,
                              "textAlign": "left",
                            }
                          }
                          title="Name">
                          <div
                            onClick={[Function]} />
                        </th>
                      </TableHeaderColumn>
                      <TableHeaderColumn
                        className=""
                        columnClassName=""
                        columnTitle={false}
                        dataAlign="left"
                        dataFormat={[Function]}
                        dataSort={false}
                        editable={true}
                        filterFormatted={false}
                        headerTitle={true}
                        hidden={false}
                        hiddenOnInsert={false}
                        isKey={false}
                        onSort={[Function]}
                        searchable={true}
                        sortIndicator={true}
                        width={null}>
                        <th
                          className=""
                          onClick={[Function]}
                          style={
                            Object {
                              "display": null,
                              "textAlign": "left",
                            }
                          }
                          title="Mode">
                          <div
                            onClick={[Function]} />
                        </th>
                      </TableHeaderColumn>
                      <TableHeaderColumn
                        className=""
                        columnClassName=""
                        columnTitle={false}
                        dataAlign="left"
                        dataFormat={[Function]}
                        dataSort={false}
                        editable={true}
                        filterFormatted={false}
                        headerTitle={true}
                        hidden={false}
                        hiddenOnInsert={false}
                        isKey={false}
                        onSort={[Function]}
                        searchable={true}
                        sortIndicator={true}
                        width={null}>
                        <th
                          className=""
                          onClick={[Function]}
                          style={
                            Object {
                              "display": null,
                              "textAlign": "left",
                            }
                          }
                          title="Duration">
                          <div
                            onClick={[Function]} />
                        </th>
                      </TableHeaderColumn>
                      <TableHeaderColumn
                        className=""
                        columnClassName=""
                        columnTitle={false}
                        dataAlign="left"
                        dataFormat={[Function]}
                        dataSort={false}
                        editable={true}
                        filterFormatted={false}
                        headerTitle={true}
                        hidden={false}
                        hiddenOnInsert={false}
                        isKey={false}
                        onSort={[Function]}
                        searchable={true}
                        sortIndicator={true}
                        width={null}>
                        <th
                          className=""
                          onClick={[Function]}
                          style={
                            Object {
                              "display": null,
                              "textAlign": "left",
                            }
                          }
                          title="Distance">
                          <div
                            onClick={[Function]} />
                        </th>
                      </TableHeaderColumn>
                      <TableHeaderColumn
                        className=""
                        columnClassName=""
                        columnTitle={false}
                        dataAlign="left"
                        dataFormat={[Function]}
                        dataSort={false}
                        editable={true}
                        filterFormatted={false}
                        headerTitle={true}
                        hidden={false}
                        hiddenOnInsert={false}
                        isKey={false}
                        onSort={[Function]}
                        searchable={true}
                        sortIndicator={true}
                        width={null}>
                        <th
                          className=""
                          onClick={[Function]}
                          style={
                            Object {
                              "display": null,
                              "textAlign": "left",
                            }
                          }
                          title="Cost">
                          <div
                            onClick={[Function]} />
                        </th>
                      </TableHeaderColumn>
                      <TableHeaderColumn
                        className=""
                        columnClassName=""
                        columnTitle={false}
                        dataAlign="left"
                        dataFormat={[Function]}
                        dataSort={false}
                        editable={true}
                        filterFormatted={false}
                        headerTitle={true}
                        hidden={false}
                        hiddenOnInsert={false}
                        isKey={false}
                        onSort={[Function]}
                        searchable={true}
                        sortIndicator={true}
                        width={null}>
                        <th
                          className=""
                          onClick={[Function]}
                          style={
                            Object {
                              "display": null,
                              "textAlign": "left",
                            }
                          }
                          title="Ridematches">
                          <div
                            onClick={[Function]} />
                        </th>
                      </TableHeaderColumn>
                    </tr>
                  </thead>
                </table>
              </div>
            </TableHeader>
            <TableBody
              bodyContainerClass={null}
              bordered={true}
              cellEdit={
                Object {
                  "__onCompleteEdit__": [Function],
                  "afterSaveCell": undefined,
                  "beforeSaveCell": undefined,
                  "blurToSave": false,
                  "mode": "none",
                }
              }
              columns={
                Array [
                  Object {
                    "align": "left",
                    "className": "",
                    "columnTitle": false,
                    "customEditor": undefined,
                    "editable": true,
                    "export": undefined,
                    "filterFormatted": false,
                    "filterValue": undefined,
                    "format": undefined,
                    "formatExtraData": undefined,
                    "hidden": true,
                    "hiddenOnInsert": false,
                    "index": 0,
                    "name": "commuterId",
                    "searchable": true,
                    "sort": false,
                    "sortFunc": undefined,
                    "sortFuncExtraData": undefined,
                    "text": undefined,
                    "width": null,
                  },
                  Object {
                    "align": "left",
                    "className": "",
                    "columnTitle": false,
                    "customEditor": undefined,
                    "editable": true,
                    "export": undefined,
                    "filterFormatted": false,
                    "filterValue": undefined,
                    "format": [Function],
                    "formatExtraData": undefined,
                    "hidden": false,
                    "hiddenOnInsert": false,
                    "index": 1,
                    "name": undefined,
                    "searchable": true,
                    "sort": false,
                    "sortFunc": undefined,
                    "sortFuncExtraData": undefined,
                    "text": "Name",
                    "width": null,
                  },
                  Object {
                    "align": "left",
                    "className": "",
                    "columnTitle": false,
                    "customEditor": undefined,
                    "editable": true,
                    "export": undefined,
                    "filterFormatted": false,
                    "filterValue": undefined,
                    "format": [Function],
                    "formatExtraData": undefined,
                    "hidden": false,
                    "hiddenOnInsert": false,
                    "index": 2,
                    "name": undefined,
                    "searchable": true,
                    "sort": false,
                    "sortFunc": undefined,
                    "sortFuncExtraData": undefined,
                    "text": "Mode",
                    "width": null,
                  },
                  Object {
                    "align": "left",
                    "className": "",
                    "columnTitle": false,
                    "customEditor": undefined,
                    "editable": true,
                    "export": undefined,
                    "filterFormatted": false,
                    "filterValue": undefined,
                    "format": [Function],
                    "formatExtraData": undefined,
                    "hidden": false,
                    "hiddenOnInsert": false,
                    "index": 3,
                    "name": undefined,
                    "searchable": true,
                    "sort": false,
                    "sortFunc": undefined,
                    "sortFuncExtraData": undefined,
                    "text": "Duration",
                    "width": null,
                  },
                  Object {
                    "align": "left",
                    "className": "",
                    "columnTitle": false,
                    "customEditor": undefined,
                    "editable": true,
                    "export": undefined,
                    "filterFormatted": false,
                    "filterValue": undefined,
                    "format": [Function],
                    "formatExtraData": undefined,
                    "hidden": false,
                    "hiddenOnInsert": false,
                    "index": 4,
                    "name": undefined,
                    "searchable": true,
                    "sort": false,
                    "sortFunc": undefined,
                    "sortFuncExtraData": undefined,
                    "text": "Distance",
                    "width": null,
                  },
                  Object {
                    "align": "left",
                    "className": "",
                    "columnTitle": false,
                    "customEditor": undefined,
                    "editable": true,
                    "export": undefined,
                    "filterFormatted": false,
                    "filterValue": undefined,
                    "format": [Function],
                    "formatExtraData": undefined,
                    "hidden": false,
                    "hiddenOnInsert": false,
                    "index": 5,
                    "name": undefined,
                    "searchable": true,
                    "sort": false,
                    "sortFunc": undefined,
                    "sortFuncExtraData": undefined,
                    "text": "Cost",
                    "width": null,
                  },
                  Object {
                    "align": "left",
                    "className": "",
                    "columnTitle": false,
                    "customEditor": undefined,
                    "editable": true,
                    "export": undefined,
                    "filterFormatted": false,
                    "filterValue": undefined,
                    "format": [Function],
                    "formatExtraData": undefined,
                    "hidden": false,
                    "hiddenOnInsert": false,
                    "index": 6,
                    "name": undefined,
                    "searchable": true,
                    "sort": false,
                    "sortFunc": undefined,
                    "sortFuncExtraData": undefined,
                    "text": "Ridematches",
                    "width": null,
                  },
                ]
              }
              condensed={false}
              data={
                Array [
                  Object {
                    "bike": Object {
                      "cost": 0.12,
                      "distance": 23,
                      "polygon": "encoded",
                      "time": 2345,
                    },
                    "car": Object {
                      "cost": 6.78,
                      "distance": 18,
                      "polygon": "encoded",
                      "time": 1234,
                    },
                    "commuterId": "1",
                    "mostLikely": Object {
                      "cost": 3.45,
                      "distance": 30,
                      "mode": "transit",
                      "polygon": "encoded",
                      "time": 3456,
                    },
                    "transit": Object {
                      "cost": 3.45,
                      "distance": 30,
                      "polygon": "encoded",
                      "time": 3456,
                    },
                    "walk": Object {
                      "cost": 0,
                      "distance": 19,
                      "polygon": "encoded",
                      "time": 6789,
                    },
                  },
                ]
              }
              hover={false}
              keyField="commuterId"
              onRowClick={[Function]}
              onRowDoubleClick={[Function]}
              onRowMouseOut={[Function]}
              onRowMouseOver={[Function]}
              onSelectRow={[Function]}
              selectRow={
                Object {
                  "bgColor": "rgb(238, 193, 213)",
                  "clickToSelect": true,
                  "mode": "radio",
                  "onSelect": [Function],
                }
              }
              selectedRowKeys={Array []}
              striped={false}
              style={
                Object {
                  "height": "100%",
                  "maxHeight": undefined,
                }
              }
              tableBodyClass={null}
              trClassName="">
              <div
                className="react-bs-container-body"
                style={
                  Object {
                    "height": "100%",
                    "maxHeight": undefined,
                  }
                }>
                <table
                  className="table table-bordered">
                  <colgroup>
                    <col
                      style={
                        Object {
                          "minWidth": 30,
                          "width": 30,
                        }
                      } />
                    <col
                      className=""
                      style={
                        Object {
                          "display": "none",
                        }
                      } />
                    <col
                      className=""
                      style={
                        Object {
                          "display": null,
                        }
                      } />
                    <col
                      className=""
                      style={
                        Object {
                          "display": null,
                        }
                      } />
                    <col
                      className=""
                      style={
                        Object {
                          "display": null,
                        }
                      } />
                    <col
                      className=""
                      style={
                        Object {
                          "display": null,
                        }
                      } />
                    <col
                      className=""
                      style={
                        Object {
                          "display": null,
                        }
                      } />
                    <col
                      className=""
                      style={
                        Object {
                          "display": null,
                        }
                      } />
                  </colgroup>
                  <tbody>
                    <TableRow
                      className=""
                      enableCellEdit={false}
                      isSelected={false}
                      onRowClick={[Function]}
                      onRowDoubleClick={[Function]}
                      onRowMouseOut={[Function]}
                      onRowMouseOver={[Function]}
                      onSelectRow={[Function]}
                      selectRow={
                        Object {
                          "bgColor": "rgb(238, 193, 213)",
                          "clickToSelect": true,
                          "mode": "radio",
                          "onSelect": [Function],
                        }
                      }
                      unselectableRow={false}>
                      <tr
                        className=""
                        onClick={[Function]}
                        onDoubleClick={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        style={
                          Object {
                            "backgroundColor": null,
                          }
                        }>
                        <TableColumn
                          className=""
                          dataAlign="center"
                          hidden={false}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": null,
                                "textAlign": "center",
                              }
                            }>
                            <input
                              checked={false}
                              disabled={false}
                              onChange={[Function]}
                              type="radio" />
                          </td>
                        </TableColumn>
                        <TableColumn
                          cellEdit={
                            Object {
                              "__onCompleteEdit__": [Function],
                              "afterSaveCell": undefined,
                              "beforeSaveCell": undefined,
                              "blurToSave": false,
                              "mode": "none",
                            }
                          }
                          className=""
                          columnTitle={null}
                          dataAlign="left"
                          hidden={true}
                          onEdit={[Function]}
                          width={null}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": "none",
                                "textAlign": "left",
                              }
                            }
                            title={null}>
                            1
                          </td>
                        </TableColumn>
                        <TableColumn
                          cellEdit={
                            Object {
                              "__onCompleteEdit__": [Function],
                              "afterSaveCell": undefined,
                              "beforeSaveCell": undefined,
                              "blurToSave": false,
                              "mode": "none",
                            }
                          }
                          className=""
                          columnTitle={null}
                          dataAlign="left"
                          hidden={false}
                          onEdit={[Function]}
                          width={null}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title={null}>
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "Luke Warm",
                                }
                              } />
                          </td>
                        </TableColumn>
                        <TableColumn
                          cellEdit={
                            Object {
                              "__onCompleteEdit__": [Function],
                              "afterSaveCell": undefined,
                              "beforeSaveCell": undefined,
                              "blurToSave": false,
                              "mode": "none",
                            }
                          }
                          className=""
                          columnTitle={null}
                          dataAlign="left"
                          hidden={false}
                          onEdit={[Function]}
                          width={null}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title={null}>
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "transit",
                                }
                              } />
                          </td>
                        </TableColumn>
                        <TableColumn
                          cellEdit={
                            Object {
                              "__onCompleteEdit__": [Function],
                              "afterSaveCell": undefined,
                              "beforeSaveCell": undefined,
                              "blurToSave": false,
                              "mode": "none",
                            }
                          }
                          className=""
                          columnTitle={null}
                          dataAlign="left"
                          hidden={false}
                          onEdit={[Function]}
                          width={null}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title={null}>
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "2 days, 9 hours, 36 minutes",
                                }
                              } />
                          </td>
                        </TableColumn>
                        <TableColumn
                          cellEdit={
                            Object {
                              "__onCompleteEdit__": [Function],
                              "afterSaveCell": undefined,
                              "beforeSaveCell": undefined,
                              "blurToSave": false,
                              "mode": "none",
                            }
                          }
                          className=""
                          columnTitle={null}
                          dataAlign="left"
                          hidden={false}
                          onEdit={[Function]}
                          width={null}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title={null}>
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "30 miles",
                                }
                              } />
                          </td>
                        </TableColumn>
                        <TableColumn
                          cellEdit={
                            Object {
                              "__onCompleteEdit__": [Function],
                              "afterSaveCell": undefined,
                              "beforeSaveCell": undefined,
                              "blurToSave": false,
                              "mode": "none",
                            }
                          }
                          className=""
                          columnTitle={null}
                          dataAlign="left"
                          hidden={false}
                          onEdit={[Function]}
                          width={null}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title={null}>
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "$3.45",
                                }
                              } />
                          </td>
                        </TableColumn>
                        <TableColumn
                          cellEdit={
                            Object {
                              "__onCompleteEdit__": [Function],
                              "afterSaveCell": undefined,
                              "beforeSaveCell": undefined,
                              "blurToSave": false,
                              "mode": "none",
                            }
                          }
                          className=""
                          columnTitle={null}
                          dataAlign="left"
                          hidden={false}
                          onEdit={[Function]}
                          width={null}>
                          <td
                            className=""
                            style={
                              Object {
                                "display": null,
                                "textAlign": "left",
                              }
                            }
                            title={null}>
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": undefined,
                                }
                              } />
                          </td>
                        </TableColumn>
                      </tr>
                    </TableRow>
                  </tbody>
                </table>
              </div>
            </TableBody>
          </div>
        </div>
      </BootstrapTable>
    </div>
  </Col>
</div>
`;
